pkg/component/gardeneradmissioncontroller/admissioncontroller.go

```diff
@@ -0,0 +1,126 @@
+// Copyright 2023 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gardeneradmissioncontroller
+
+import (
+	"context"
+	"time"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/gardener/gardener/pkg/component"
+	operatorclient "github.com/gardener/gardener/pkg/operator/client"
+	"github.com/gardener/gardener/pkg/utils/flow"
+	"github.com/gardener/gardener/pkg/utils/managedresources"
+	secretsmanager "github.com/gardener/gardener/pkg/utils/secrets/manager"
+)
+
+const (
+	managedResourceNameRuntime = "gardener-admission-controller-runtime"
+	managedResourceNameVirtual = "gardener-admission-controller-virtual"
+)
+
+// TimeoutWaitForManagedResource is the timeout used while waiting for the ManagedResources to become healthy or
+// deleted.
+var TimeoutWaitForManagedResource = 5 * time.Minute
+
+// Values contains configuration values for the gardener-admission-controller resources.
+type Values struct {
+}
+
+// New creates a new instance of DeployWaiter for the gardener-admission-controller.
+func New(client client.Client, namespace string, secretsManager secretsmanager.Interface, values Values) component.DeployWaiter {
+	return &admissioncontroller{
+		client:         client,
+		namespace:      namespace,
+		secretsManager: secretsManager,
+		values:         values,
+	}
+}
+
+type admissioncontroller struct {
+	client         client.Client
+	namespace      string
+	secretsManager secretsmanager.Interface
+	values         Values
+}
+
+func (a admissioncontroller) Deploy(ctx context.Context) error {
```

Comment:
Any particular reason why you do not use a pointer as we mostly do?

(Applies to the whole file and its siblings.)

```suggestion
func (a *admissioncontroller) Deploy(ctx context.Context) error {
```

Reply:
I didn't use it because it was simply not required. After the feedback from @rfranzke I changed it however to make it consistent, probably missed the adaption in this file 🙂